app/controllers/reports_controller.rb

```diff
@@ -20,6 +46,26 @@ def export
 
   private
 
+  def report_params
+    return params unless params[:report]
+    params.require(:report).permit(:start_date, :end_date, client_ids: [], project_ids: [], task_ids: [], user_ids: [])
+  end
+
+  def set_form_data
+    @selectable_clients = Client.all
+    @selectable_projects = Project.all
+    @selectable_tasks = Task.all
+    @selectable_users = User.all
+
+    @selected_start_date = Date.parse(report_params[:start_date]) if report_params[:start_date].present?
+    @selected_end_date = Date.parse(report_params[:end_date]) if report_params[:end_date].present?
+
+    @selected_client_ids = report_params[:client_ids].to_a.map(&:to_i)
+    @selected_project_ids = report_params[:project_ids].to_a.map(&:to_i)
+    @selected_user_ids = report_params[:user_ids].to_a.map(&:to_i)
+    @selected_task_ids = report_params[:task_ids].to_a.map(&:to_i)
```

Comment:
Let's put all of these form data variables in a datastructure, so we only have to pass one variable to the partial.

Example:
```ruby
@form_data = OpenStruct.new(
    selectable_clients: Client.all,
    selectable_projects: Project.all,
    ...etc
  )
```

It will make it easier to understand that all of these variables are form-related